### ISSUE TITLE
pkginfo causing build issues with webpack

### ISSUE THREAD

**Issue by DavidArayan** [seanfisher/passport-microsoft#25]
Hello,

We are getting the following error when trying to bundle code using webpack. We also use `passport-local` and `passport-google-oauth20` strategies with no issues. 

We think this is due to `pkginfo` trying to find a `package.json` when trying to be bundled/combined with bundlers like webpack. The root cause seems to be this line of code at [index.js#L10](https://github.com/seanfisher/passport-microsoft/blob/0456aa9bce05579c18e77f51330176eb26373658/lib/index.js#L10)

`
Error: Could not find package.json up from ./node_modules/pkginfo/lib/pkginfo.js
`

For now we fixed the issue in webpack by loading this module externally as follows

```json
"externals": {
        "passport-microsoft": {
                 "commonjs": "passport-microsoft"
        }
}
```

It would be great to resolve this issue so `passport-microsoft` can be bundled.

**Comment by JoelKlint** [comment]
I'm sharing this issue when bundling with `esbuild`. 

What is the purpose of this line? It seems like it assigns `module.version`, but I do not understand the gain in doing so

**Comment by JoelKlint** [comment]
PR opened 👍 
@DavidArayan

**Comment by kfarkasHU** [comment]
For anyone who is facing with issue (like us), and scratches their head for so long:

`package.json` changes:
```json
"scripts": {
 "postinstall": "node .tools/apply-patches.mjs"
}
```

`.tools/apply-patches.mjs` content (excerpt):
```js

import fs from "fs";
import path from 'path';
import { fileURLToPath } from 'url';

export const __filename = fileURLToPath(import.meta.url);
export const __dirname = path.dirname(__filename);

const packagePath = "node_modules/passport-microsoft/lib";
const fileName = "index.js";

const removeStrings = [
    "require('pkginfo')(module, 'version')"
];

const fullFilePath = path.join(__dirname, "..", packagePath, fileName);
let content = fs.readFileSync(fullFilePath).toString();

removeStrings.forEach(m => content = content.replace(m, ''));

fs.writeFileSync(fullFilePath, content);
console.log("passport-microsoft has been patched");

```

After you run `npm install` or `yarn` this magic will remove the line from the given file what causes the issue. Once the related PR is merged to this repository (and it will be published), this code will do nothing at all, and will cause no issues.
In defense of this solution i'd like mention, that we had to do the same kind of hack with `when` because their dependency (`vertx`) has been removed from npmjs (and has been republished under a different name).

**Comment by seanfisher** [comment]
Merged #27, which fixes this issues. Planning on releasing updated version today with this fix and others.

Thanks to all for your input and sharing workarounds in the meantime!
